cmd/coreinstance/create.go

```diff
@@ -1,19 +1,30 @@
 package coreinstance
 
 import (
+	"fmt"
 	"github.com/spf13/cobra"
 
 	cfg "github.com/calyptia/cli/config"
 )
 
 func NewCmdCreateCoreInstance(config *cfg.Config) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "core_instance",
-		Short: "Setup a new core instance on either Kubernetes, Amazon EC2, or Google Compute Engine",
+		Use:               "core_instance",
+		Short:             "Setup a new core instance on either Kubernetes, Amazon EC2, or Google Compute Engine",
+		PersistentPreRunE: checkForProjectToken(config),
 	}
 	cmd.AddCommand(newCmdCreateCoreInstanceOnK8s(config, nil))
 	cmd.AddCommand(newCmdCreateCoreInstanceOnAWS(config, nil, nil))
 	cmd.AddCommand(newCmdCreateCoreInstanceOnGCP(config, nil))
 	cmd.AddCommand(newCmdCreateCoreInstanceOperator(config, nil))
 	return cmd
 }
+
+func checkForProjectToken(config *cfg.Config) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if config.ProjectToken == "" {
```

Comment:
This can be done with [MarkPersistentFlagRequired](https://pkg.go.dev/github.com/spf13/cobra#Command.MarkPersistentFlagRequired) But this can do it too.

Reply:
No, it's not the same. Since you can persist a token with the `config set_token` command instead of passing the flag each time.